In [ ]:
# Exercise 1:

In [ ]:
import pandas as pd
from scipy import stats

data = {
    "Test_Case": ["T1", "T2", "T3", "T4", "T5", "T6", "T7", "T8"],
    "Response_Time_Before": [420, 450, 410, 470, 440, 460, 430, 445],
    "Response_Time_After": [360, 390, 355, 400, 380, 395, 370, 385]
}

df = pd.DataFrame(data)
df

,Test_Case,Response_Time_Before,Response_Time_After
0,T1,420,360
1,T2,450,390
2,T3,410,355
3,T4,470,400
4,T5,440,380
5,T6,460,395
6,T7,430,370
7,T8,445,385


In [ ]:
Two_Sample_Results1=stats.ttest_rel(
    df["Response_Time_Before"],
    df["Response_Time_After"]
)
Two_Sample_Results1

TtestResult(statistic=np.float64(39.08847772565582), pvalue=np.float64(1.8674871068411005e-09), df=np.int64(7))

Since the t-calc = 39.08 is greater than the t-table = 2.365 and the p-value is less than α = 0.05, the result is statistically significant.

In [ ]:
# Exercise 2:

In [ ]:
import pandas as pd
from scipy import stats

data = {
    "Household_ID": ["H1","H2","H3","H4","H5","H6","H7","H8","H9","H10"],
    "Daily_Energy_Consumption": [11.2, 10.8, 11.5, 10.6, 11.0, 11.3, 10.9, 11.1, 10.7, 11.4]
}

df = pd.DataFrame(data)
df

,Household_ID,Daily_Energy_Consumption
0,H1,11.2
1,H2,10.8
2,H3,11.5
3,H4,10.6
4,H5,11.0
5,H6,11.3
6,H7,10.9
7,H8,11.1
8,H9,10.7
9,H10,11.4


In [ ]:
one_sample_result=stats.ttest_1samp(
    df["Daily_Energy_Consumption"],
    11
)
one_sample_result

TtestResult(statistic=np.float64(0.5222329678671007), pvalue=np.float64(0.6141172548083891), df=np.int64(9))

Since t-calc = 0.52 is less than t-table = 2.262, and the p-value is greater than α = 0.05, the result is not statistically significant.
Hence, the null hypothesis is accepted and the alternative hypothesis is rejected.